diff --git a/node_modules/@haqq/encryption-react-native/HaqqEncryptionRN.podspec b/node_modules/@haqq/encryption-react-native/HaqqEncryptionRN.podspec
index c12800e..f7b3077 100644
--- a/node_modules/@haqq/encryption-react-native/HaqqEncryptionRN.podspec
+++ b/node_modules/@haqq/encryption-react-native/HaqqEncryptionRN.podspec
@@ -17,7 +17,7 @@ Pod::Spec.new do |s|
   s.source_files = "ios/**/*.{h,m,mm,swift}"
 
   s.dependency "React-Core"
-  s.dependency "CryptoSwift", "1.6.0"
+  s.dependency "CryptoSwift", "1.5.1"
 
   # Don't install the dependencies when we run `pod install` in the old architecture.
   if ENV['RCT_NEW_ARCH_ENABLED'] == '1' then
