diff --git a/node_modules/@sentry/react-native/sentry.gradle b/node_modules/@sentry/react-native/sentry.gradle
index b4671fa..f1f5ea7 100644
--- a/node_modules/@sentry/react-native/sentry.gradle
+++ b/node_modules/@sentry/react-native/sentry.gradle
@@ -118,7 +118,7 @@ gradle.projectsEvaluated {
 
               def resolvedCliPackage = null
               try {
-                  resolvedCliPackage = new File(["node", "--print", "require.resolve('@sentry/cli/package.json')"].execute(null, rootDir).text.trim()).getParentFile();
+                  resolvedSentryPath = new File("$reactRoot/node_modules/@sentry/react-native/package.json").getParentFile();
               } catch (Throwable ignored) {}
               def cliPackage = resolvedCliPackage != null && resolvedCliPackage.exists() ? resolvedCliPackage.getAbsolutePath() : "$reactRoot/node_modules/@sentry/cli"
               def cliExecutable = sentryProps.get("cli.executable", "$cliPackage/bin/sentry-cli")
@@ -168,7 +168,7 @@ gradle.projectsEvaluated {
 
               def resolvedSentryPath = null
               try {
-                  resolvedSentryPath = new File(["node", "--print", "require.resolve('@sentry/react-native/package.json')"].execute(null, rootDir).text.trim()).getParentFile();
+                  resolvedSentryPath = new File("$reactRoot/node_modules/@sentry/react-native/package.json").getParentFile();
               } catch (Throwable ignored) {} // if the resolve fails we fallback to the default path
               def sentryPackage = resolvedSentryPath != null && resolvedSentryPath.exists() ? resolvedSentryPath.getAbsolutePath() : "$reactRoot/node_modules/@sentry/react-native"
 
