diff --git a/node_modules/bn.js/lib/bn.js b/node_modules/bn.js/lib/bn.js
index adecc94..0da27e6 100644
--- a/node_modules/bn.js/lib/bn.js
+++ b/node_modules/bn.js/lib/bn.js
@@ -52,10 +52,25 @@
   try {
     if (typeof window !== 'undefined' && typeof window.Buffer !== 'undefined') {
       Buffer = window.Buffer;
-    } else {
+    }
+
+    if (global.Buffer && typeof Buffer === 'undefined') {
+      Buffer = global.Buffer;
+    }
+
+    if(typeof Buffer === 'undefined') {
       Buffer = require('buffer').Buffer;
     }
+
+    if(typeof Buffer === 'undefined') {
+      Buffer = require('buffer/').Buffer;
+    }
+
+    if(typeof Buffer === 'undefined') {
+      throw new Error('Buffer not supported');
+    }
   } catch (e) {
+    console.log(e.message);
   }
 
   BN.isBN = function isBN (num) {
