diff --git a/node_modules/lottie-react-native/src/android/build.gradle b/node_modules/lottie-react-native/src/android/build.gradle
index 2d63065..c35ca13 100644
--- a/node_modules/lottie-react-native/src/android/build.gradle
+++ b/node_modules/lottie-react-native/src/android/build.gradle
@@ -3,7 +3,7 @@ def safeExtGet(prop, fallback) {
 }
 
 apply plugin: 'com.android.library'
-apply from: 'gradle-maven-push.gradle'
+// apply from: 'gradle-maven-push.gradle'
 
 //def DEFAULT_ANDROID_SUPPORT_LIB_VERSION = "1.0.0"
 
