diff --git a/node_modules/pbkdf2/browser.js b/node_modules/pbkdf2/browser.js
index e07b5ad..453c918 100644
--- a/node_modules/pbkdf2/browser.js
+++ b/node_modules/pbkdf2/browser.js
@@ -15,14 +15,7 @@ exports.pbkdf2 = function (password, salt, iterations, keylen, digest, callback)
   })
 }
 
-var defaultEncoding
-if (process.browser) {
-  defaultEncoding = 'utf-8'
-} else {
-  var pVersionMajor = parseInt(process.version.split('.')[0].slice(1), 10)
-
-  defaultEncoding = pVersionMajor >= 6 ? 'utf-8' : 'binary'
-}
+var defaultEncoding = 'utf-8';
 
 exports.pbkdf2Sync = function (password, salt, iterations, keylen, digest) {
   if (!Buffer.isBuffer(password)) password = new Buffer(password, defaultEncoding)
