diff --git a/node_modules/react-native-encrypted-storage/android/build.gradle b/node_modules/react-native-encrypted-storage/android/build.gradle
index b343dc8..96cede0 100644
--- a/node_modules/react-native-encrypted-storage/android/build.gradle
+++ b/node_modules/react-native-encrypted-storage/android/build.gradle
@@ -124,7 +124,7 @@ dependencies {
 
   testImplementation 'junit:junit:4.13.1'
 
-  androidTestImplementation 'androidx.test.ext:junit:1.1.2'
+  androidTestImplementation 'androidx.test.ext:junit:1.1.5'
   //noinspection GradleDependency
   androidTestImplementation 'org.mockito:mockito-android:3.4.6'
   androidTestImplementation 'androidx.test.espresso:espresso-core:3.3.0'
