diff --git a/node_modules/react-native-reanimated/android/build.gradle b/node_modules/react-native-reanimated/android/build.gradle
index da46b94..f5adaec 100644
--- a/node_modules/react-native-reanimated/android/build.gradle
+++ b/node_modules/react-native-reanimated/android/build.gradle
@@ -243,9 +243,9 @@ def JS_RUNTIME = {

     // Check if Hermes is enabled in app setup
     def appProject = rootProject.allprojects.find { it.plugins.hasPlugin('com.android.application') }
-    if ((REACT_NATIVE_MINOR_VERSION >= 71 && appProject?.hermesEnabled?.toBoolean()) || appProject?.ext?.react?.enableHermes?.toBoolean()) {
-        return "hermes"
-    }
+  if (REACT_NATIVE_MINOR_VERSION >= 71 && appProject?.ext?.react?.enableHermes?.toBoolean()) {
+    return "hermes"
+  }

     // Use JavaScriptCore (JSC) by default
     return "jsc"
