diff --git a/node_modules/realm/react-native/android/src/main/java/io/realm/react/RealmReactModule.java b/node_modules/realm/react-native/android/src/main/java/io/realm/react/RealmReactModule.java
index e45e03e..19d1af2 100644
--- a/node_modules/realm/react-native/android/src/main/java/io/realm/react/RealmReactModule.java
+++ b/node_modules/realm/react-native/android/src/main/java/io/realm/react/RealmReactModule.java
@@ -248,13 +248,13 @@ class RealmReactModule extends ReactContextBaseJavaModule {
             final String json = map.get("postData");
             if (json == null) {
                 Response response = newFixedLengthResponse("");
-                response.addHeader("Access-Control-Allow-Origin", "http://localhost:8081");
+                response.addHeader("Access-Control-Allow-Origin", "*");
                 return response;
             }
             final String jsonResponse = processChromeDebugCommand(cmdUri, json);
 
             Response response = newFixedLengthResponse(jsonResponse);
-            response.addHeader("Access-Control-Allow-Origin", "http://localhost:8081");
+            response.addHeader("Access-Control-Allow-Origin", "*");
             return response;
         }
     }
